### PR TITLE
Committed

### PR DESCRIPTION
**Pull request by dheer2005**

### CODE REVIEW

File: Async_Await.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            {
+                var result1 = awaiter1.GetResult();
+                Calculate2(result1);
+            });
+
+            Calculate3();
+            
+            /*Task.WaitAll(task1,task2);
+            var awaiter1 = task1.GetAwaiter();
+            var awaiter2 = task2.GetAwaiter();
```
Comment (RoshanFS) [inline Async_Await.cs]: Remove commented lines

File: Async_Await2.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        public static async Task StartSchoolAssembly()
+        {
+            await Task.Run(() =>
+            {
+                Thread.Sleep(5000);
+                Console.WriteLine("School Started");
+            });
+            }
```
Comment (RoshanFS) [inline Async_Await2.cs]: make proper indentation

File: BankImplementation.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+namespace Abstraction
+{
+    //interface
+    internal class BankImplementation : Bank, Promotion
+    {
+
+        int balance = 20000;
+        public void CheckBalance()
+        {
+            if (this.Login(1111))
```
Comment (RoshanFS) [inline BankImplementation.cs]: Do not use hardcoded value

File: BankImplementation.cs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        internal int balance = 12000;
+        public override void CheckBalance()
+        {
+            if (this.Login(1111))
+            {
+                Console.WriteLine("Balance is {0}", balance);
+            }
+            else
+            {
+                Console.WriteLine("Invalid Pin Code");
```
Comment (RoshanFS) [inline BankImplementation.cs]: Remove it

File: BillingNew.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Abstraction
+{
+    internal class BillingNew:Billing
```
Comment (RoshanFS) [inline BillingNew.cs]: Make proper indentation

File: BillingNew.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        public override float calcBill(int finalamount)
+        {
+            return finalamount + (finalamount * 0.18F);
+        }
+
+        /*public static void Main(string[] args)
+        {
+            //method overhiding
+            Billing obj = new BillingNew();
+            Console.WriteLine("Due to Method overhiding total bill is: {0}",obj.calcBill(100));//output will 100 = thats why this is called method overhiding
```
Comment (RoshanFS) [inline BillingNew.cs]: Remove it

File: DelegateExample.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+            Console.WriteLine("Function Body");
+        }
+
+        public void fun1()
+        {
+            Console.WriteLine("Second Function Body");
+        }
+
+        public delegate void myDel();
+
```
Comment (RoshanFS) [inline DelegateExample.cs]: I hope that you understood the concept of delegate

File: Employee5.cs [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            this.Id = Id;   
+            this.Name = Name;
+        }
+
+        //parameterless constructor for deserialization
+        public Employee2() { }
+    }
+}
+
```
Comment (RoshanFS) [inline Employee5.cs]: Why there are a lot of lines left blank

File: EventExample.cs [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace fundamentals
+{
+    internal class EventExample
```
Comment (RoshanFS) [inline EventExample.cs]: Are we using this class anywhere?

Comment (dheer2005) [reply EventExample.cs]: Sir, I did not understand this concept well, I am doing it now.

File: Inheritance.cs [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    public override double Area()
+    {
+        return Math.Pow(2, Length);
+    }
+}
+public class Rectangle : Drawing
+{
+    public double Height { get; set; }
+    public double Width { get; set; }
+    public Rectangle()
```
Comment (RoshanFS) [inline Inheritance.cs]: Make proper indentation

File: JsonSerialize.cs [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+            Console.WriteLine("File created Successfully -> " + path);*/
+
+        /* //JSON Serialization
+
+             string path = @"D:\Serailization_DemoFile\sample1.json";
+             Employee2 emp = new Employee(242, "Raja");
+
+             // Serialize to JSON
+             string jsonString = JsonSerializer.Serialize(emp);
+
```
Comment (RoshanFS) [inline JsonSerialize.cs]: Remove commented code

File: Serialization_Demo.csproj [hunk 12]
```diff
@@ -0,0 +1,10 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
```
Comment (RoshanFS) [inline Serialization_Demo.csproj]: Do not push this file

File: fundamentals.csproj [hunk 13]
```diff
@@ -0,0 +1,11 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (RoshanFS) [inline fundamentals.csproj]: Do not push this code

**Pull request closed without merging**